graph/src/components/ethereum/adapter.rs

```diff
@@ -360,15 +383,15 @@ impl From<EthereumBlockFilter> for EthereumCallFilter {
             contract_addresses_function_signatures: ethereum_block_filter
                 .contract_addresses
                 .into_iter()
-                .map(|address| (address, HashSet::default()))
-                .collect::<HashMap<Address, HashSet<[u8; 4]>>>(),
+                .map(|(start_block_opt, address)| (address, (start_block_opt, HashSet::default())))
+                .collect::<HashMap<Address, (u64, HashSet<[u8; 4]>)>>(),
```

Comment:
This will take the last `start_block` value, but the correct behaviour is the minimum.